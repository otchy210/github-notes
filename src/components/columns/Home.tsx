import React from 'react';
import { useGitHub } from '../../providers/GitHubProvider';
import { HeaderAndNav } from '../common/HeaderAndNav';
import { Column } from './Column';
import { List } from '../contents/List';
import { Guide } from '../contents/Guide';

export const Home: React.FC = () => {
  const { repoStatus, client } = useGitHub();
  return (
    <Column>
      <HeaderAndNav />
      <article className="home">
        {repoStatus === 'unknown' ? (
          <div className="icon loading">
            <img src="/images/icon-circle.svg" />
          </div>
        ) : client ? (
          <List />
        ) : (
          <Guide />
        )}
      </article>
    </Column>
  );
};
