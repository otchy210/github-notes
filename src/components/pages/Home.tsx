import React from 'react';
import { useGitHub } from '../../providers/GitHubProvider';
import { Guide } from './Guide';
import { List } from './List';

export const Home: React.FC = () => {
  const { client } = useGitHub();
  return <>{client ? <List /> : <Guide />}</>;
};
