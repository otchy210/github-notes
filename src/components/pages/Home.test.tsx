import React from 'react';
import { render } from '@testing-library/react';
import { Home } from './Home';
import { MemoryRouter } from 'react-router-dom';

describe('Home', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, { wrapper: MemoryRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});
