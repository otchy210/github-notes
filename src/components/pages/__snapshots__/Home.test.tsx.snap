// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    GitHub Notes
  </h2>
  <p>
    GitHub Notes is a tool that open source version of Apple Notes or Google Keep which stores notes in your GitHub repository. So it is actually depending on Microsoft but having much better control. All your notes are written as markdown documents.
  </p>
  <p>
    You need to create a personal token via 
    <a
      href="https://github.com/settings/tokens/new"
      target="_blank"
    >
      this page
    </a>
     to use this tool. And the access token should have "repo" permission which provides full control of private repositories. So you may be worried if this tool does something bad. In that case, you can always check the 
    <a
      href="https://github.com/otchy210/github-notes"
      target="_blank"
    >
      source code
    </a>
     of this tool and confirm there is no problem.
  </p>
  <p>
    Did you create your access token? Set it up in the 
    <a
      href="/config"
    >
      config
    </a>
     page!
  </p>
</DocumentFragment>
`;
